### DISCUSSION TITLE
Memory available to MBasic

### DISCUSSION THREAD

**Discussion by laffineuse** [thehighnibble/thehighnibble.github.io#110] [IMSAI8080esp replica]
I am trying to run the program wargames.bas from Retro Tech Shed and even boot from their CP/M wargames.dsk.  However when entering MBasic they show 34782 bytes free and I only show 24632, leading to MBasic running out of memory when loading the .bas file.  How can I make more memory available to Mbasic?

**Comment by donfroula** [comment]
Does entering "CLEAR , 34782" help? 

Don


===============

CLEAR 

Format ; CLEAR [, [expression1][ , expression2]]

expression1 is a memory location which, if specified, sets the highest location available for use by BASIC-80.

expression2 sets aside stack space for BASIC. The default is 256 bytes or one-eighth of the available memory, whichever is smaller.

Examples : 
CLEAR 
CLEAR , 32768 
CLEAR , ,2000
CLEAR ,32768,2000

**Comment by nbreeden2** [comment]
How much memory does the CPM configuration show? It will show the amount of RAM it's configured for during boot. CPM is configured for a fixed amount of memory. 64K of RAM in the systems doesn't mean CPM knows about all the RAM; if the configuration is for 48K then that's all the RAM CPM will see. I'm not sure how MBASIC determines the total amount of RAM but my guess is it queries CPM for the value.
I believe the STAT command will also show the free RAM assuming STAT.COM is on your boot floppy.
For CPM to see a different amount of ram it needs to be manually configured.
